### PR TITLE
add command to let admins send custom cargo orders

### PR DESCRIPTION
**Pull request by Chief-Engineer**
## About the PR

Lets admins add custom cargo orders. As implemented, the order item is teleported onto the shuttle when it is delivered so it needs to be staged somewhere safe, like an admin testing area.

**Media**

image
image

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**

:cl:
- add: Admins can now send custom cargo orders.

### CODE REVIEW

File: Content.Shared/Cargo/CargoOrderData.cs [hunk 1]
```diff
@@ -16,6 +16,11 @@ public sealed class CargoOrderData
         /// </summary>
         public readonly int OrderId;
 
+        /// <summary>
+        /// Entity to use as the order. Overrides <see cref="ProductId"/> and <see cref="OrderQuantity"/>
+        /// </summary>
+        public readonly EntityUid? OrderEntity;
+
```
Comment (metalgearsloth) [inline Content.Shared/Cargo/CargoOrderData.cs:23]: Adding a bunch of special-casing around this seems kinda weird to me.

Comment (metalgearsloth) [reply Content.Shared/Cargo/CargoOrderData.cs:23]: I think that's probably better yeah.

File: Content.Client/Cargo/UI/CargoConsoleMenu.xaml.cs
Comment (AJCM-git) [inline Content.Client/Cargo/UI/CargoConsoleMenu.xaml.cs:1]: As far as i can tell the only difference between the null and not null parts of the main part of this are the `Icon` field, cant you use a ternary operator or something to make this a lot cleaner (or make the command always send a default icon or something)?

File: Content.Client/Cargo/UI/CargoShuttleMenu.xaml.cs
Comment (AJCM-git) [inline Content.Client/Cargo/UI/CargoShuttleMenu.xaml.cs:1]: same

File: Content.Server/Administration/Commands/Station/AddCustomCargoOrder.cs [hunk 2]
```diff
@@ -0,0 +1,71 @@
+﻿using Content.Server.Cargo.Components;
+using Content.Server.Cargo.Systems;
+using Content.Shared.Administration;
+using Robust.Shared.Console;
+
+namespace Content.Server.Administration.Commands.Station;
+
+[AdminCommand(AdminFlags.Fun)]
+public sealed class AddCustomCargoOrder : LocalizedCommands
```
Comment (AJCM-git) [inline Content.Server/Administration/Commands/Station/AddCustomCargoOrder.cs:9]: can you update this to the new `ToolshedCommand` API

File: Content.Server/Administration/Logs/Converters/CargoOrderDataStringRepresentationConverter.cs [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+public sealed class CargoOrderDataStringRepresentationConverter : AdminLogConverter<CargoOrderStringRepresentation>
+{
+    public override void Write(Utf8JsonWriter writer, CargoOrderStringRepresentation value, JsonSerializerOptions options)
+    {
+        writer.WriteStartObject();
+
+        writer.WriteNumber("orderId", value.OrderId);
+        writer.WriteString("productName", value.ProductName);
+        if (value.EntityUid != null)
+            writer.WriteNumber("entityUid", value.EntityUid.Value.GetHashCode()); // How do I make this use EntityUidConverter.cs instead?
```
Comment (AJCM-git) [inline Content.Server/Administration/Logs/Converters/CargoOrderDataStringRepresentationConverter.cs:16]: make a WriteEntityUid method? (if thats the problem, idk if thats your actual issue here)

File: Content.Shared/Cargo/Orders/CargoOrderDataEntity.cs
Comment (AJCM-git) [inline Content.Shared/Cargo/Orders/CargoOrderDataEntity.cs:1]: filescoped namespace

**Review by AJCM-git — CHANGES_REQUESTED** [review]
Looks mostly good

**Comment by moonheart08** [comment]
@Chief-Engineer i hate to do this to you but needs rewritten to use toolshed if you ever come back to it.

**Pull request closed without merging**